bsmetadata/train.py

```diff
@@ -156,10 +195,26 @@ def main(args: CFG) -> None:
     else:
         args.num_train_epochs = math.ceil(args.max_train_steps / num_update_steps_per_epoch)
 
-    if args.num_eval < 1:
+    if args.evaluation_strategy == IntervalStrategy.EPOCH:
+        if args.eval_num_per_epoch < 1:
+            eval_per_n_step = args.max_train_steps + 1
+        else:
+            eval_per_n_step = args.max_train_steps // args.eval_num_per_epoch
+    elif args.evaluation_strategy == IntervalStrategy.STEPS:
+        eval_per_n_step = args.eval_steps
+    else:  # IntervalStrategy.NO
         eval_per_n_step = args.max_train_steps + 1
-    else:
-        eval_per_n_step = args.max_train_steps // args.num_eval
+
+    if args.save_strategy == IntervalStrategy.EPOCH:
+        if args.save_num_per_epoch < 1:
+            save_per_n_step = args.max_train_steps + 1
+        else:
+            save_per_n_step = args.max_train_steps // args.save_num_per_epoch
```

Comment:
(nit)
This can be re-written to:
```
if (args.save_strategy == IntervalStrategy.EPOCH 
   and args.save_num_per_epoch >= 1):
    save_per_n_step = args.max_train_steps // args.save_num_per_epoch
```
so there is fewer lines and fewer `+1`

Reply:
It's a good idea, I pushed the change